src/Compiler/Optimize/LowerComputedCollections.fs

```diff
@@ -393,14 +393,7 @@ module Array =
                     mkCompGenLetIn m (nameof count) (tyOfExpr g count) count (fun (_, count) ->
                         let countTy = tyOfExpr g count
 
-                        // count < 1
-                        let countLtOne =
-                            if isSignedIntegerTy g countTy then
-                                mkILAsmClt g m count (mkTypedOne g m countTy)
-                            else
-                                mkAsmExpr ([AI_clt_un], [], [count; mkTypedOne g m countTy], [g.bool_ty], m)
-
-                        // if count < 1 then
+                        // if count = 0 then
```

Comment:
Just for my understanding - why was it there and why is it not needed anymore?

Reply:
An earlier approach I took to compute the count meant that it could sometimes be negative — e.g., $count = (finish - start) / step + 1$ when $finish - start > 0$ and $step < 0$ — so checking that $count < 1$ was the simplest way of testing whether we could just return an empty array or whether we needed to initialize one.

With the current way the count is computed, it can never be negative — it always uses an unsigned type anyway.

Reply:
Alrighty!